### PR TITLE
Use rrcl node instead of clrc+rrc

### LINKED ISSUE #3: SelectionDAG fails on assertion
Input for llc:
```
target datalayout = "e-m:e-p:16:16-i32:16-i64:16-f32:16-f64:16-a:8-n8:16-S16"
target triple = "msp430-elf"
define i16 @cttz(i16 %i) nounwind readnone {
entry:
  %0 = tail call i16 @llvm.cttz.i16(i16 %i, i1 true)
  ret i16 %0
}
declare i16 @llvm.cttz.i16(i16, i1) nounwind readnone speculatable
```
Message:
```
llc: /home/mskvortsov/llvm/lib/CodeGen/SelectionDAG/ScheduleDAGSDNodes.cpp:353: void llvm::ScheduleDAGSDNodes::BuildSchedUnits(): Assertion `N->getNodeId() == -1 && "Node already inserted!"' failed.
Stack dump:
0.	Program arguments: /home/mskvortsov/llvm-build/bin/llc cttz.ll 
1.	Running pass 'Function Pass Manager' on module 'cttz.ll'.
2.	Running pass 'MSP430 DAG->DAG Pattern Instruction Selection' on function '@cttz'
```

### PR DESCRIPTION
**Pull request by mskvortsov**
This fixes #3.

### CODE REVIEW

File: src/llvm/lib/Target/MSP430/MSP430InstrInfo.td [hunk 1]
```diff
@@ -215,7 +215,8 @@ let isCodeGenOnly = 1, usesCustomInserter = 1 in {
   def Srl16    : Pseudo<(outs GR16:$dst), (ins GR16:$src, GR8:$cnt),
                         "# Srl16 PSEUDO",
                         [(set GR16:$dst, (srl GR16:$src, GR8:$cnt))]>;
-  def Clrc     : Pseudo<(outs), (ins), "", [(MSP430clrc)]>;
+  def Rrl      : Pseudo<(outs GR16:$dst), (ins GR16:$src), "",
```
Comment (asl) [inline src/llvm/lib/Target/MSP430/MSP430InstrInfo.td]: Should we indicate the we implicitly use / def status register here?

File: src/llvm/lib/Target/MSP430/MSP430ISelLowering.h [hunk 2]
```diff
@@ -36,8 +36,8 @@ namespace llvm {
       /// Y = RRC X, rotate right via carry
       RRC,
 
-      /// Clear carry flag
-      CLRC,
+      /// Rotate right via carry, carry is set to zero beforehand
```
Comment (asl) [inline src/llvm/lib/Target/MSP430/MSP430ISelLowering.h]: I would prefer slightly more "sound" name. So one won't confuse with usual shift.

Comment (mskvortsov) [reply src/llvm/lib/Target/MSP430/MSP430ISelLowering.h]: `rra` = rotate right arithmetically
`rrl` = rotate right logically which sounds good enough to me
Maybe just a comment change will suffice?

Comment (asl) [reply src/llvm/lib/Target/MSP430/MSP430ISelLowering.h]: But we have side effect with clearing the carry. So if we'd try to start using this we need to take this into account. So I'd at least keep "C" in the name :)

File: src/llvm/lib/Target/MSP430/MSP430InstrInfo.td (hunk 1 above)
Comment (mskvortsov) [reply src/llvm/lib/Target/MSP430/MSP430InstrInfo.td]: `let Def = [SR]` is here already. And we don't use SR, just clear carry bit.

File: src/llvm/lib/Target/MSP430/MSP430ISelLowering.h (hunk 2 above)
Comment (mskvortsov) [reply src/llvm/lib/Target/MSP430/MSP430ISelLowering.h]: Reasonable, will do.

**Pull request merged**